### PR TITLE
[Bug]Fix some schema change not work right

### PR DESCRIPTION
**Pull request by WingsGo**
This CL including:
1. Fix numeric type to varchar type not work right bug
2. remove convert from largeint type to double type in doc.
3. change logic check in FE to support convert decimal type to varchar type which has already been implement in BE.

### CODE REVIEW

**Comment by WingsGo** [comment]
For #4007

**Comment by chaoyli** [comment]
It's better add more discrete message in commit info like you issue.

**Comment by WingsGo** [comment]
> It's better add more discrete message in commit info like you issue.

Ok, done, and I tested the schema change supported in doc, but I found that convert LargetInt to Double is not support in code implement, maybe you can confirm that why it is added to doc in case It is supported but there is some bug exists?

**Comment by WingsGo** [comment]
In order to make review easiler, I will not add code format until review ended.

File: be/src/olap/schema_change.cpp [hunk 1]
```diff
@@ -194,6 +194,64 @@ ColumnMapping* RowBlockChanger::get_mutable_column_mapping(size_t column_index)
...
+ConvertTypeResolver::ConvertTypeResolver() {
+    // supported type convert should annotate in doc:
+    // http://doris.incubator.apache.org/master/zh-CN/sql-reference/sql-statements/Data%20Definition/ALTER%20TABLE.html#description
+    // from varchar type
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_TINYINT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_SMALLINT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_INT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_BIGINT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_LARGEINT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_FLOAT>();
```
Comment (morningman) [inline be/src/olap/schema_change.cpp:231]: is VARCHAR to DOUBLE and DATETIME support?

**Review by morningman — COMMENTED** [review]
comment

File: be/src/olap/schema_change.cpp (hunk 1 above)
Comment (WingsGo) [reply be/src/olap/schema_change.cpp:232]: VAHRCHAR->DOUBLE is supported. Support VARCHAR->DATETIME maybe supported in next PR.

**Comment by WingsGo** [comment]
I have tested by following:

```
CREATE TABLE `test_schema_change` (
	`k1` int NULL,
	`v1` varchar NULL
) ENGINE=OLAP
DUPLICATE KEY(`k1`)
DISTRIBUTED BY HASH(`k1`) BUCKETS 1
PROPERTIES (
 "replication_num" = "1"
);
alter table test_schema_change modify column v1 tinyint;
alter table test_schema_change modify column v1 varchar;
alter table test_schema_change modify column v1 smallint;
alter table test_schema_change modify column v1 varchar;
alter table test_schema_change modify column v1 int;
alter table test_schema_change modify column v1 varchar;
alter table test_schema_change modify column v1 bigint;
alter table test_schema_change modify column v1 varchar;
alter table test_schema_change modify column v1 largeint;
alter table test_schema_change modify column v1 varchar;
alter table test_schema_change modify column v1 float;
alter table test_schema_change modify column v1 varchar;
alter table test_schema_change modify column v1 double;
alter table test_schema_change modify column v1 varchar;
alter table test_schema_change modify column v1 date;
```

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by chaoyli** [comment]
> > It's better add more discrete message in commit info like you issue.
> 
> Ok, done, and I tested the schema change supported in doc, but I found that convert LargetInt to Double is not support in code implement, maybe you can help me confirm that why it is added to doc in case It is supported but there is some bug exists?

I test the case, it's forbidden by FE to change LARGEINT to DOUBLE. Because LARGEINT is int128_t and double is 2^52 and will lose information.
So you can remove from the doc.

**Comment by WingsGo** [comment]
> > > It's better add more discrete message in commit info like you issue.
> > 
> > 
> > Ok, done, and I tested the schema change supported in doc, but I found that convert LargetInt to Double is not support in code implement, maybe you can help me confirm that why it is added to doc in case It is supported but there is some bug exists?
> 
> I test the case, it's forbidden by FE to change LARGEINT to DOUBLE. Because LARGEINT is int128_t and double is 2^52 and will lose information.
> So you can remove from the doc.

Ok, I removed it, thx~

**Comment by morningman** [comment]
@chaoyli  Any comments?

**Review by chaoyli — APPROVED** [review]

**Pull request merged**